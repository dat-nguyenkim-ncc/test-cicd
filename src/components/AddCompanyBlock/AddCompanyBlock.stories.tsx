import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import AddCompanyBlock, { AddCompanyBlockProps } from './AddCompanyBlock'

export default {
  title: 'Widgets/AddCompanyBlock',
  component: AddCompanyBlock,
} as Meta

const Template: Story<AddCompanyBlockProps> = args => <AddCompanyBlock {...args} />

export const Default = Template.bind({})
Default.args = {} as AddCompanyBlockProps
