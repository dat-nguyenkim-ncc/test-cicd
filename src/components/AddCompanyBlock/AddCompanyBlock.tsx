import React, { useRef } from 'react'
import { Flex } from 'theme-ui'
import { acceptedFormats } from '../../utils'
import { Button } from '../'
import { Heading, Paragraph, Section } from '../primitives'
import strings from '../../strings'
import { ViewInterface } from '../../types'

export type AddCompanyBlockProps = ViewInterface<{
  onPressForm(): void
}>

const AddCompanyBlock = ({ onPressForm, sx }: AddCompanyBlockProps) => {
  const {
    pages: { addCompany: copy },
  } = strings

  const refFile = useRef<HTMLInputElement | null>(null)

  const onFileChange = () => {
    if (refFile.current?.files?.length) {
      console.log(refFile.current.files)
    }
  }

  return (
    <>
      <input
        ref={refFile}
        type="file"
        multiple
        hidden
        accept={acceptedFormats.csv.toString()}
        onChange={onFileChange}
        style={{ display: 'none' }}
      />

      <Section sx={sx}>
        <Flex sx={{ justifyContent: 'center' }}>
          <Flex
            sx={{
              justifyContent: 'center',
              borderRadius: 10,
              py: 5,
              px: 6,
              flexDirection: 'column',
              backgroundColor: 'gray02',
              width: `calc(50% - 30px)`,
            }}
          >
            <Heading center sx={{ mb: 4 }} as="h4">
              {copy.options.form.title}
            </Heading>
            <Paragraph center sx={{ mb: 4 }}>
              {copy.options.form.body}
            </Paragraph>
            <Button onPress={onPressForm} variant="primary" label={copy.options.form.button} />
          </Flex>
        </Flex>
      </Section>
    </>
  )
}

export default AddCompanyBlock
