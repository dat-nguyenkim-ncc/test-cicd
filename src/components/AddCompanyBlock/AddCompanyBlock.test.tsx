import * as React from 'react'
import { shallow } from 'enzyme'
import AddCompanyBlock from './AddCompanyBlock'

describe('AddCompanyBlock', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<AddCompanyBlock onPressForm={() => {}} />)
    expect(wrapper).toMatchSnapshot()
  })
})
