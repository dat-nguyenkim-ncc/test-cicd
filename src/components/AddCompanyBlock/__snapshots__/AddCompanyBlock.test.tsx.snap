// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCompanyBlock matches snapshot 1`] = `ShallowWrapper {}`;
